lib/auth/v4/awsURIencode.js

```diff
@@ -36,7 +36,7 @@ function awsURIencode(input, encodeSlash, noEncodeStar) {
     const encSlash = encodeSlash === undefined ? true : encodeSlash;
     let encoded = '';
     for (let i = 0; i < input.length; i++) {
-        const ch = input.charAt(i);
+        var ch = input.charAt(i);
```

Comment:
I'd rather use `let` as this is a local variable.

Reply:
Are you ok with me force pushing this change or would you prefer individual commits and squash when merging?

Reply:
Yes, force push is good